(Many) Patches

TheRSO opened the pull request:
## Loadout panel changes
### Before (2560 x 1440 (16:9) only)
![](https://media.discordapp.net/attachments/991465089544761427/1313095024522821692/image.png?ex=674ee289&is=674d9109&hm=fba3c85c973086b8318ab4712bce78dd23e16877df9a42c07a1e24829cd8676d&=&format=webp&quality=lossless&width=802&height=451)![](https://media.discordapp.net/attachments/991465089544761427/1313095110090821675/image.png?ex=674ee29e&is=674d911e&hm=ec862ec34bd1fd05fcf040e837b16c24bd03faffb8672dbefd5ba4c6db6d12ab&=&format=webp&quality=lossless&width=802&height=451)

### After
|  | **2560 x 1440 (16:9)** | **1440 x 1080 (4:3)** |
|--|--|--|
|  | ![](https://media.discordapp.net/attachments/991465089544761427/1313130076325281866/image.png?ex=674f032e&is=674db1ae&hm=a1244d22822edd4978d97067437a05aaad66edd12a6937d9889e9227f78e542d&=&format=webp&quality=lossless&width=802&height=451) | ![](https://media.discordapp.net/attachments/991465089544761427/1313131402543366164/image.png?ex=674f046b&is=674db2eb&hm=535c38edaa4146edd78890b2322db60841868b1d63f6a4781b00326b39cc86e2&=&format=webp&quality=lossless&width=802&height=602) |
|  | ![](https://media.discordapp.net/attachments/991465089544761427/1313130128125198336/image.png?ex=674f033b&is=674db1bb&hm=2c55015cf1c4ac8a8535fcd15a05c27d2daa823610f75141e705023d91452ec7&=&format=webp&quality=lossless&width=550&height=309) | ![](https://media.discordapp.net/attachments/991465089544761427/1313131445966864384/image.png?ex=674f0475&is=674db2f5&hm=b3584dacb0ef9d42cda19d565e67e9411f17d324de573f0f9f4b42b098d52a67&=&format=webp&quality=lossless&width=802&height=602) |
|  | ![](https://media.discordapp.net/attachments/991465089544761427/1313130404919775243/image.png?ex=674f037d&is=674db1fd&hm=ce76af533ee0bec15684c73991fa6c5d7e24a848f0c7622a3b43b1a1e50603e2&=&format=webp&quality=lossless&width=802&height=451) | ![](https://media.discordapp.net/attachments/991465089544761427/1313131937954660402/image.png?ex=674f04ea&is=674db36a&hm=7275f49abd212a3642e3bd68e2b3f65e2a97405c9fbe48f042056854d073e1f3&=&format=webp&quality=lossless&width=802&height=602) |


> [!NOTE] 
> Tried to align loadout and lower buttons using xpos p.[float] and regualar xpos, but couldnt align on both resolutions. The space in between the loadout buttons and from the left of the preset panel does not seem to behave nice. Button width stays unchanged between both resolutions.
## Medigun/vacc charge moved down

### Before
![](https://media.discordapp.net/attachments/991465089544761427/1306586531770667008/image.png?ex=674e4748&is=674cf5c8&hm=b86745ebffe48814efe7c4435d080c66171408b25da897da18f3998734dd5a83&=&format=webp&quality=lossless)
![](https://media.discordapp.net/attachments/991465089544761427/1306586746229620746/image.png?ex=674e477b&is=674cf5fb&hm=92adae9d39a598dde6f06079ca9c4e4e85071960de26e07365044d4e2c4c2010&=&format=webp&quality=lossless)

### After
![](https://media.discordapp.net/attachments/991465089544761427/1313126875660423178/image.png?ex=674f0033&is=674daeb3&hm=043cfff776ad57a181d3b88894e6b8ce82514b9749458589836a4a867bfef4e5&=&format=webp&quality=lossless)
![](https://media.discordapp.net/attachments/991465089544761427/1313126971047149598/image.png?ex=674f004a&is=674daeca&hm=f87acccf74dc0fb5c9987bb7ecf7eeebc55a3f9b21379754d72687d4f8ce481d&=&format=webp&quality=lossless)

## Top bar changes
### Nudged top bar icons
**Left is after, Right is before**
![](https://media.discordapp.net/attachments/991465089544761427/1313124535087923220/image.png?ex=674efe05&is=674dac85&hm=ff87825b5856adf80b1c6e1de4d5c6c25e88ef09f2040a704e6ede4f671f0699&=&format=webp&quality=lossless&width=1253&height=384)
### Removed gradient, replace with plain bg
**Top is Before, bottom is after**
![](https://media.discordapp.net/attachments/991465089544761427/1313125054095032360/image.png?ex=674efe81&is=674dad01&hm=36de595ac1315277bba8ebfaece406cd51360836c47c2944ebde94eab49175f1&=&format=webp&quality=lossless&width=1253&height=120)**Contrast boosted**
![](https://media.discordapp.net/attachments/991465089544761427/1313129675366731837/image.png?ex=674f02cf&is=674db14f&hm=a142093e941b5aa9f9c1ed4896f64744b2d8d68ddf8971a851c6773d4ead383c&=&format=webp&quality=lossless&width=1253&height=278)
## Other
|  | Before | After |
|--|--|--|
|**Moved party chat left**| ![](https://cdn.discordapp.com/attachments/991465089544761427/1308037190214881300/image.png?ex=674e4850&is=674cf6d0&hm=b943d0a3575cec5b98f780447d7a812b6ea0e823810a8ca1d6b2affdd5346e6e) | ![](https://cdn.discordapp.com/attachments/991465089544761427/1313118405124952075/image.png?ex=674ef850&is=674da6d0&hm=323177d072a986cf33b69fc99ebe32f471b7f5927645c793547c350c39b7f555&) |
| **Brightened separator bar** | ![](https://media.discordapp.net/attachments/991465089544761427/1313096419682422804/image.png?ex=674ee3d6&is=674d9256&hm=c227e9f30310712715cc4c98eb8dea189bb9885fa3664107a7e7609c69a1ac3c&=&format=webp&quality=lossless) | ![](https://media.discordapp.net/attachments/991465089544761427/1313122234079903804/image.png?ex=674efbe1&is=674daa61&hm=0657036e18e0d1866c097eb6d5ec53e6655da9ec473a6382a5818df10d85a8e0&=&format=webp&quality=lossless) |
|**Nudged FL**|![](https://media.discordapp.net/attachments/991465089544761427/1313095684932505611/image.png?ex=674ee327&is=674d91a7&hm=90b6918d8671744225a3ea53052a377752a7c3235381079b9880cdc80ed6ca7c&=&format=webp&quality=lossless)|![](https://media.discordapp.net/attachments/991465089544761427/1313156031127228496/image.png?ex=674f1b5b&is=674dc9db&hm=ef008258c16c0890b82e9e1253fa21c4ff58f7110b38f1f5d0663a85b1eb7a72&=&format=webp&quality=lossless)|
| **Moved voices button next to text chat** | ![](https://media.discordapp.net/attachments/991465089544761427/1313118893727809577/image.png?ex=674ef8c4&is=674da744&hm=c42fa034d8ee2834939676780a6f5b670d3be61c62ae47da77ba22b6a52fc4ff&=&format=webp&quality=lossless&width=839&height=562) | ![](https://media.discordapp.net/attachments/991465089544761427/1313122019440722010/image.png?ex=674efbae&is=674daa2e&hm=0931f25af9a9d4f9e7b3bfc3002666356a7c9dd6bfba20aef0d0d3fa8ea221fe&=&format=webp&quality=lossless&width=837&height=562) |
| **Removed quit dialog border** | ![](https://media.discordapp.net/attachments/991465089544761427/1313133678557138944/image.png?ex=674f0689&is=674db509&hm=864aab4f1a34d14bcb6692c538f06dbf09401035460daee52be8590b49928c03&=&format=webp&quality=lossless) | ![](https://media.discordapp.net/attachments/991465089544761427/1313135043379462175/image.png?ex=674f07cf&is=674db64f&hm=51f123ee024652235ff573c6555d995baab3cab654f9773b370fc421dcb6f9a8&=&format=webp&quality=lossless) |
| **Nudged Killstreak Icon** | ![](https://media.discordapp.net/attachments/991465089544761427/1313162996771455019/image.png?ex=674f21d7&is=674dd057&hm=2b6269671c819266cbd457ead55c1a62e8d5237e68dbb58ae89293abb0c36361&=&format=webp&quality=lossless) | ![`](https://media.discordapp.net/attachments/991465089544761427/1313127341634879508/image.png?ex=674f00a2&is=674daf22&hm=dab780f721fe8007980da6c2f48bf5c1b4890ba4945b69ea50fd7e645fda4469&=&format=webp&quality=lossless) |
| **Hotfix for black bars** | ![](https://media.discordapp.net/attachments/991465089544761427/1313099111771410503/image.png?ex=674ee658&is=674d94d8&hm=cdbc19be9b0a69b41eb8412fb655c64e11b37d8e0caac4565e0afe62eb58931c&=&format=webp&quality=lossless&width=350&height=580) | ![](https://media.discordapp.net/attachments/991465089544761427/1313125958391300126/image.png?ex=674eff59&is=674dadd9&hm=3ee2657422673d0ef7bb0535d93dd2600f598852a1fab40673a9d166ea12caca&=&format=webp&quality=lossless&width=395&height=580) |
|  | | |




> [!NOTE] 
> Most of these changes were also checked on 4:3, but for sake of time/space, screenshots not included
>
> Pictures not to scale
>
> Reloaded game between resolution changes (launch options)

TheRSO committed:
Fixed Elements Cutting Off [classloadoutpanel.res]

Fixed class model cutting off on the left side
Fixed attributes label cutting off and reduced font size

TheRSO committed:
Adjusted main menu party chat [globalchat.res]

Fixed it being offset by 5 units

TheRSO committed:
Brightened seperator bar [links.res]

29 29 29  255 -> 44 44 44 255

TheRSO committed:
Removed Gradient on top bar [matchmakingdashboard]

Replaced gradient with a bg that only covers party member icons on the main menu top bar. The gradient was banding.

TheRSO committed:
Moved icons down slightly [matchmakingdashboard.res]

TheRSO committed:
Nudged friendslist 1unit [mainmenuoverride.res]

TheRSO committed:
Nudged Killstreak Icon on [targetid.res]

TheRSO committed:
Refixed medigun charges colliding with mvm Sheild hudmediccharge.res

TheRSO committed:
Potiential fix for black bars [hudweaponselection]

Taken from huds tf discord user Selbi
https://discord.com/channels/673304949718908948/1138517261561364612/1138517261561364612
changing width/height of model in any way seems to break it

TheRSO committed:
Update hudweaponselection.res

TheRSO committed:
Moved voice button next to chat button [tools.res]

TheRSO committed:
Moved voice button next to chat button [tools.res]

TheRSO committed:
"Ah shit"-Critical flaw

TheRSO committed:
Removed border from quit dialog [confirmdialog.res]

TheRSO committed:
nudged friends list closer to center [mainmenuoverride.res]

TheRSO committed:
Merge branch 'master' into patch

CriticalFlaw commented on resource/ui/classloadoutpanel.res:
The `PassiveAttributes` change looks good but the class model changes don't translate well to lower 4:3 resolutions:
![20241207172443_1](https://github.com/user-attachments/assets/2cb46fc2-d9cd-4e9f-b3ff-ac3c8a3fa8ef)

What is the reason for updating the model position from the way it is today?

TheRSO replied:
2560x1440
![image](https://github.com/user-attachments/assets/40c82a6d-6116-473b-ac43-13750713ebcb)
[more on pr comment]

The current version uses Xpos to move it right, this has the consequence of making the player model clip off on the left side when you rotate it/or if the class model is big.

I was having a hard time finding something that worked well for 16:9 and 4:3 since 4:3 shrinks the width and thereoff the model size

TheRSO replied:
Will look into further when I get home.

TheRSO replied:
Off topic, but your 4:3 looks slightly dif 
(1440 x 1080)
![image](https://github.com/user-attachments/assets/3c10864d-1278-4232-8bf8-34119c39006e)

TheRSO replied:
For the class model
```"wide"     "p[float]"``` --> 4:3 width is too small makes class model also small or 16:9 too big
```"wide"     "[int]"``` --> panel extends too far right interferes with buttons or 16:9 player model becomes small

also tried playing with model ```origin_xpos```, but that might be in HU, I'm not exactly sure, but it doesn't seem to scale with resolution.

prob will end up using custom class data due to pyro being extremely wide with stock flame compared to the tiny scout

TheRSO replied:
@CriticalFlaw pushed another commit not sure how you feel but I sized pyro small so it wouldn't clip off if a player rotated him. 4:3 kind of forces me to have pretty limited space to work with on 16:9
this is 2560 x 1440 (16:9)
![image](https://github.com/user-attachments/assets/8c1734b6-7da0-445f-bb52-e9ea08f5a7b1)

1440 x 1080 (4:3)
![image](https://github.com/user-attachments/assets/0e8214af-1139-48d9-8463-87551b29e579)

CriticalFlaw replied:
It appears too far the larger the resolution is. Maybe we revisit this particular one in a separate PR and merge the rest?
![20241208215440_1](https://github.com/user-attachments/assets/3234339c-518a-4bf6-af82-2ade77a7fe13)

TheRSO replied:
can pin the panel to an anchor with p[float] as its xpos
```"xpos"     "c-[int]"``` on class model causes the model to stretch horizontally
there will have to be some level of compromise.

[pyro clips left side a bit, but fov can be lowered]



**HYPOTHETICAL not committed yet**
## 2560 x 1080 21:9
![image](https://github.com/user-attachments/assets/5eb43b51-1b00-4e36-a774-3e857ed53e1c)
![image](https://github.com/user-attachments/assets/ed5debf8-cc84-4da6-901f-2cc635ac4add)
![image](https://github.com/user-attachments/assets/c14fbc06-5738-4e63-99dd-f3f490d0a4e4)


## 2560 x 1440 16:9
![image](https://github.com/user-attachments/assets/e657878f-72d8-4718-9fd0-644792118559)
![image](https://github.com/user-attachments/assets/b01b8f60-51ee-4e01-abda-e761466341cb)
![image](https://github.com/user-attachments/assets/0c28063e-0311-49a6-8b83-c15c9f13ba28)


## 1440 x 1080 4:3
note: this panel behaves dif for whatever reason on 4:3 when I have this funk setup
![image](https://github.com/user-attachments/assets/5967d37a-be68-472e-b149-cb05e289a045)
![image](https://github.com/user-attachments/assets/f4eb341d-045e-4dde-81a9-dbdf9280829e)
![image](https://github.com/user-attachments/assets/872b4af6-8235-4a1f-8b26-63dd88c450e0)

## 1280 x 1080 5:4
it functions ig, prob what most of 5:4 will end up being atm
![image](https://github.com/user-attachments/assets/59fe1541-d76f-403e-b30b-6e23655bd36b)
![image](https://github.com/user-attachments/assets/68d3e163-725a-4b64-bc8c-ba72e81fe846)
![image](https://github.com/user-attachments/assets/5becf466-b213-4bfd-b164-3b05e53b4764)
![image](https://github.com/user-attachments/assets/cc9a34e7-c52c-45da-9a6d-6b1981b6418c)

CriticalFlaw replied:
The screenshots look pretty good to me. Could you commit that to this PR and I'll do a quick test after work?

TheRSO replied:
@CriticalFlaw done

CriticalFlaw commented on resource/ui/hudweaponselection.res:
Can you post a screenshot of black bars before the fix and what resolution you've used to test this? I'm not able to reproduce this one on my side and don't recall it being reported before so I'm double checking to make sure this change is needed here.

TheRSO replied:
It’s at the bottom of PR comment.
![image](https://github.com/user-attachments/assets/be2c35aa-56b2-4378-8984-cea946b49d98)
It’s on 2560x1440 

1440x1080
![image](https://github.com/user-attachments/assets/dc24dd4a-36db-4aba-bd66-f8d0495366b0)

TheRSO replied:
Changes for this res file reverted. I didn't check closely enough but bigger (unskinned) weapons ie minigun were getting cut off.

TheRSO replied:
ah I changed the tall value before by accident IC. It's looks fine now (model 80 x 50).

TheRSO committed:
Reverted Changes [ hudweaponselection.res]

Non skinned minigun/others got cropped off (oops)

CriticalFlaw committed:
Added back tweaked black bar fix for weapon selection

TheRSO committed:
Merge branch 'master' into patch

TheRSO committed:
Added class specific model sizing [classloadout]

Pyro is tiny now
16:9 models are a bit more left than I prefer, but now model size should be much more consistent between 16:9 and 4:3

TheRSO committed:
Merge branch 'master' into patch

TheRSO committed:
Added Anchor for class model [classloadout]
